File: PYTHON/UDPServer.py
from socket import *
serverName = 'localhost'
serverPort = 12000
serverSocket = socket(AF_INET, SOCK_DGRAM)
serverSocket.bind(('', serverPort))
print("The server is ready to receive")
while True:
    message, clientAddress = serverSocket.recvfrom(2048)
    modifiedmessage = message.decode().upper()
    serverSocket.sendto(modifiedmessage.encode(), clientAddress)

File: RUBY/UDPServer.rb
require 'socket'

server_port = 12000

# Create a UDP socket
server_socket = UDPSocket.new

# Bind the socket to the local host and port
server_socket.bind("localhost", server_port)

puts "The server is ready to receive"

loop do
  # Receive the message from the client
  message, client_address = server_socket.recvfrom(2048)

  # Modify the message to uppercase
  modified_message = message.upcase

  # Send the modified message back to the client
  server_socket.send(modified_message, 0, client_address[3], client_address[1])
end
